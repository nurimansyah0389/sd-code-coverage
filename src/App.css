@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');

* {
    box-sizing: border-box;
}

html,
body {
    margin: 0 auto;
    font-family: 'Open Sans', sans-serif;
    font-size: 14px;
}

main {
    max-width: 480px;
    border: 1px solid #ddd;
    margin: 20px;
    padding: 20px;
}

h1 {
    margin: 0 auto;
}

label {
    display: block;
    margin-bottom: 10px;
}

input {
    padding: 10px;
    border: 1px solid #ddd;
    border-radius: 4px;
}
input + button {
    margin-left: 10px;
}

button {
    border: 1px solid #ddd;
    background: transparent;
    padding: 10px;
    border-radius: 4px;
}

ul > li {
    margin-bottom: 20px;
}
ul > li > button {
    padding: 5px;
    background: #e74c3c;
    color: white;
    border-color: transparent;
}