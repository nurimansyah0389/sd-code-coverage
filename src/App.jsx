import React, { Fragment } from 'react';

import List from './components/List';

import './App.css';

const data = ['First Todo', 'Second Todo', 'Third Todo'];

const App = () => (
    <Fragment>
        <main>
            <h1>Code Coverage Tests</h1>
            <List data={data} />
        </main>
    </Fragment>
);
export default App;