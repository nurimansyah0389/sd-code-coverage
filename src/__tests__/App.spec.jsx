import React from 'react';
import Renderer from 'react-test-renderer';

import App from '../App';

describe('App Spec', () => {
    it('Should render without error', () => {
        const tree = Renderer.create(<App/>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});