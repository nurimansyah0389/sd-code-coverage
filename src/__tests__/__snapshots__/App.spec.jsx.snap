// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Spec Should render without error 1`] = `
<main>
  <h1>
    Code Coverage Tests
  </h1>
  <div>
    <ul>
      <li>
        First Todo
         - 
        <button
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </li>
      <li>
        Second Todo
         - 
        <button
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </li>
      <li>
        Third Todo
         - 
        <button
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </li>
    </ul>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <div>
        <label>
          What will you do next?
        </label>
        <input
          id="todo"
          name="todo"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Add Todo
        </button>
      </div>
    </form>
  </div>
</main>
`;
